### PR TITLE
[mesheryctl] Split config functionality and refractor

### LINKED ISSUE #2160: [mesheryctl] Split current context functionality
**Current Behavior**

Single source of execution in root.go and create.go in context commands.

Ref #2104 

**Desired Behavior**

Split functionality into two functions, 
1. for creating the config file
2. Adding in template context into the context file

### PR DESCRIPTION
**Pull request by anirudhjain75**
Split config file creation into readable functions and refactor some code for optimization. Allow users to pass in --set flag as argument for assigning the context during creation. Also fix issue where when deleting context the name is left behind in current context

Signed-off-by: Anirudh Jain <anirudh.jain@outlook.com>

**Description**

This PR fixes 
#2160 
#2161 

**Notes for Reviewers**
You're gonna get tired 😆

### CODE REVIEW

File: mesheryctl/internal/cli/root/root.go [hunk 1]
```diff
@@ -144,36 +141,16 @@ func initConfig() {
...
-						log.Fatalln(err)
-					}
+				// Create config file if not present in meshery folder
+				err = utils.CreateConfigFile()
+				if err != nil {
+					log.Fatal(err)
+				}
+
+				// Add Context to context file
+				err = utils.AddContextToConfig("local", utils.TemplateContext, fmt.Sprintf("%s/%s", utils.MesheryFolder, "config.yaml"), true)
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/root.go:151]: Is the name of the config file a centralized (global) variable?

File: mesheryctl/internal/cli/root/system/context/create.go [hunk 2]
```diff
@@ -23,35 +20,25 @@ var createContextCmd = &cobra.Command{
...
-		configuration.CurrentContext = args[0]
-		viper.Set("contexts", configuration.Contexts)
-		viper.Set("current-context", configuration.CurrentContext)
-		err = viper.WriteConfig()
-		if err != nil {
-			log.Fatal(err)
-		}
-		log.Printf("Added %s context and set as current context", args[0])
+
+		log.Printf("Added %s context", args[0])
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/context/create.go:36]:
```suggestion
		log.Printf("Added '%s' context", args[0])
```

File: mesheryctl/internal/cli/root/system/context/create.go [hunk 3]
```diff
@@ -23,35 +20,25 @@ var createContextCmd = &cobra.Command{
...
 		return nil
 	},
 }
 
 func init() {
 	createContextCmd.Flags().StringVarP(&url, "url", "u", "https://localhost:9081", "Meshery Server URL with Port")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/context/create.go:42]: Yikes. What's this reference to localhost here?

File: mesheryctl/internal/cli/root/system/context/view.go [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
+	Args:         cobra.NoArgs,
+	SilenceUsage: true,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		err := viper.Unmarshal(&configuration)
 		if err != nil {
 			return err
 		}
 		currentContext := viper.GetString("current-context")
+		if currentContext == "" {
+			return errors.New("context not set")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/context/view.go:24]:
```suggestion
			return errors.New("current context not set")
```

File: mesheryctl/pkg/utils/helpers.go [hunk 5]
```diff
@@ -485,3 +502,70 @@ func AskForConfirmation(s string) bool {
...
+}
+
+// AddContextToConfig adds context passed to it to mesheryctl config file
+func AddContextToConfig(contextName string, context models.Context, configPath string, set bool) error {
+	var currentConfig models.MesheryCtlConfig
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
+		return err
+	}
+
+	file, err := ioutil.ReadFile(configPath)
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/helpers.go:524]: Use viper here instead?

File: mesheryctl/pkg/utils/helpers.go [hunk 6]
```diff
@@ -485,3 +502,70 @@ func AskForConfirmation(s string) bool {
...
+	if set {
+		currentConfig.CurrentContext = contextName
+	}
+
+	content, err := yaml.Marshal(currentConfig)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(configPath, content, 0644)
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/helpers.go:553]: `viper.WriteConfig()`?

File: mesheryctl/pkg/utils/helpers.go [hunk 7]
```diff
@@ -485,3 +502,70 @@ func AskForConfirmation(s string) bool {
...
+}
+
+// ValidateURL validates url provided for meshery backend to mesheryctl context
+func ValidateURL(URL string) error {
+	ParsedURL, err := url.ParseRequestURI(URL)
+	if err != nil {
+		return err
+	}
+	if ParsedURL.Scheme != "http" && ParsedURL.Scheme != "https" {
+		return fmt.Errorf("%s is not a supported request scheme", ParsedURL.Scheme)
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/helpers.go:568]:
```suggestion
		return fmt.Errorf("%s is not a supported protocol", ParsedURL.Scheme)
```

File: mesheryctl/internal/cli/root/system/context/create.go (hunk 3 above)
Comment (anirudhjain75) [reply mesheryctl/internal/cli/root/system/context/create.go:42]: If you dont pass it a url, it uses that.Also it acts as example when requesting help on cli

File: mesheryctl/pkg/utils/helpers.go (hunk 5 above)
Comment (anirudhjain75) [reply mesheryctl/pkg/utils/helpers.go:524]: Actually don't need any viper functionality here so didn't use it

File: mesheryctl/internal/cli/root/root.go (hunk 1 above)
Comment (anirudhjain75) [reply mesheryctl/internal/cli/root/root.go:151]: No not yet.. that was planned for I think.. I'll make it one though 👍

Comment (anirudhjain75) [reply mesheryctl/internal/cli/root/root.go]: The commit I just pushed addresses this issue but calls for another issue. The AuthTokenLocation is hard to determine. Need to shift the Local template to after the function for setting absolute path is called

File: mesheryctl/internal/cli/root/system/context/create.go (hunk 3 above)
Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/context/create.go:42]: Ok

File: mesheryctl/pkg/utils/helpers.go (hunk 5 above)
Comment (leecalcote) [reply mesheryctl/pkg/utils/helpers.go:527]: Got it. This concerns me though in that lack of consistently breeds bugs.

**Review by leecalcote — APPROVED** [review]
Let's give this a shot.

**Pull request merged**